frame/support/procedural/src/lib.rs

```diff
@@ -583,3 +584,643 @@ pub fn storage_alias(_: TokenStream, input: TokenStream) -> TokenStream {
 		.unwrap_or_else(|r| r.into_compile_error())
 		.into()
 }
+
+/// Used internally to decorate pallet attribute macro stubs when they are erroneously used
+/// outside of a pallet module
+fn pallet_macro_stub() -> TokenStream {
+	quote!(compile_error!("This attribute can only be used within a `pallet` module")).into()
```

Comment:
I think it's more accurate to say that it can only be used within a module with the `#[frame_support::pallet]` annotation.

Reply:
done, thanks!